api/src/main/java/org/openmrs/module/addresshierarchy/service/AddressHierarchyServiceImpl.java

```diff
@@ -705,7 +706,15 @@ else if (level.getParent() == null){
 	synchronized public void initializeFullAddressCache() {
 
 		// generally, this global property should be set to true; it just allows cache load to be disabled to speed startup
-		if (Context.getAdministrationService().getGlobalProperty(AddressHierarchyConstants.GLOBAL_PROP_INITIALIZE_ADDRESS_HIERARCHY_CACHE_ON_STARTUP).equalsIgnoreCase("true")) {
+		String globalProperty = "";
+		try {
+			Context.addProxyPrivilege("Get Global Properties");

```

Comment:
Don't we have a constant for the privilege?

Reply:
We'll have to update the platform version. However, it is giving me a lot of errors.

Reply:
Which version did you upgrade it to? Can i look at the errors?

Reply:
Thinking about this again, it does not look good to do this from here because this can be called from different contexts. Can we do it from where the call originates?

Reply:
I updated the PR.

Reply:
I tried updating the platform version to 1.11.6. I am seeing multiple test errors similar to this:

```
[ERROR] getChildAddressHierarchyEntries_shouldGetChildAddressHierarchyEntries(org.openmrs.module.addresshierarchy.AddressHierarchyServiceTest)  Time elapsed: 0.164 s  <<< ERROR!
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to read candidate component class: file [/home/wikum/Documents/Code/OpenMRS/openmrs-module-addresshierarchy/api/target/classes/org/openmrs/module/addresshierarchy/i18n/DisabledI18nCache.class]; nested exception is java.lang.ArrayIndexOutOfBoundsException: 1
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1

```